src/nvim/screen.c

```diff
@@ -5823,8 +5823,10 @@ int screen_valid(int doclear)
 void win_grid_alloc(win_T *wp, int doclear)
 {
   if (wp->w_grid.ScreenLines == NULL
-      || wp->w_grid.internal_Rows != wp->w_height
-      || wp->w_grid.internal_Columns != wp->w_width) {
+      // TODO(utkarshme): these lines create a redraw very frequently. find the problem
```

Comment:
The problem is that you are comparing the "outer" size, which is irrelevant here. You want to calculate `1.5 * wp->w_height` to a local variable, and then compare that to `w_grid.Rows`.

Reply:
Or maybe we should just delete the 1.5 thing, now that external size control is almost working.